src/actions/websocket.js

```diff
@@ -132,37 +115,32 @@ function doReconnect() {
         const currentChannelId = getCurrentChannelId(state);
         const currentUserId = getCurrentUserId(state);
 
-        // We try and get the posts for the current channel as soon as possible
-        if (currentChannelId) {
-            dispatch(markChannelAsRead(currentChannelId));
-            dispatch(loadPostsHelper(currentChannelId));
-        }
+        if (currentTeamId) {
+            const dmPrefs = getDirectShowPreferences(state);
+            const statusesToLoad = {
+                [currentUserId]: true,
+            };
+
+            for (const pref of dmPrefs) {
+                if (pref.value === 'true') {
+                    statusesToLoad[pref.name] = true;
+                }
+            }
 
-        dispatch(getLicenseConfig());
-        dispatch(getClientConfig());
-        dispatch(getMyTeams());
-        dispatch(getMyTeamMembers());
-        dispatch(getMyTeamUnreads());
-        dispatch(getMyPreferences());
-        dispatch(loadProfilesForDirect());
-        dispatch(getTeams());
+            dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(() => {
+                dispatch(loadProfilesForDirect());
+            });
 
-        if (currentTeamId) {
-            dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(({data}) => {
-                if (data && currentChannelId) {
-                    const {channels, members} = data;
-                    const stillMemberOfCurrentChannel = members.find((m) => m.channel_id === currentChannelId);
-
-                    if (!stillMemberOfCurrentChannel) {
-                        const defaultChannel = channels.find((c) => c.team_id === currentTeamId && c.name === General.DEFAULT_CHANNEL);
-                        if (defaultChannel) {
-                            EventEmitter.emit(General.DEFAULT_CHANNEL, defaultChannel.display_name);
-                            selectChannel(defaultChannel.id)(dispatch, getState);
-                        }
-                    }
+            dispatch(getPosts(currentChannelId)).then(({error}) => {
+                // Getting the posts return the user is not authorized, redirecting to the default channel
+                if (error && error.status_code === 403) {
```

Comment:
I don't have any specific suggestions to improve this, but something about "try getting the posts and if we get a 403, we've been removed from the channel" feels hacky. I'd prefer something more straightforward like "get the latest channels and if we're not in the current channel, we've been removed from the channel".

Reply:
agree, I'll change it.